src/components/MarkdownEditor/FloatingLinkEditorPlugin.tsx

```diff
@@ -0,0 +1,403 @@
+import styled from '@emotion/styled';
+import { colors, misc, shadows, spacing } from '@ndla/core';
+import { FieldErrorMessage, FormControl, InputV3, Label } from '@ndla/forms';
+import { ButtonV2 } from '@ndla/button';
+import { getSelectedNode } from './EditorToolbar';
+
+const VERTICAL_GAP = 10;
+const HORIZONTAL_OFFSET = 5;
+
+export const ADD_LINK_COMMAND: LexicalCommand<null> = createCommand();
+
+const FloatingContainer = styled.div`
+  position: absolute;
+  z-index: 1000;
+  display: none;
+  align-items: center;
+  gap: ${spacing.small};
+  background-color: ${colors.white};
+  border: 1px solid ${colors.brand.greyLight};
+  border-radius: ${misc.borderRadius};
+  padding: ${spacing.small};
+  box-shadow: ${shadows.levitate1};
+  &[data-visible='true'] {
+    display: flex;
+  }
+`;
+
+const InputWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${spacing.small};
+`;
+
+export const setFloatingElemPositionForLinkEditor = (
+  targetRect: DOMRect | null,
+  floatingElem: HTMLElement,
+  anchorElem: HTMLElement,
+  verticalGap: number = VERTICAL_GAP,
+  horizontalOffset: number = HORIZONTAL_OFFSET,
+) => {
+  const scrollerElem = anchorElem.parentElement;
+
+  if (targetRect === null || !scrollerElem) {
+    floatingElem.style.opacity = '0';
+    floatingElem.style.transform = 'translate(-10000px, -10000px)';
+    return;
+  }
+
+  const floatingElemRect = floatingElem.getBoundingClientRect();
+  const anchorElementRect = anchorElem.getBoundingClientRect();
+  const editorScrollerRect = scrollerElem.getBoundingClientRect();
+
+  let top = targetRect.top - verticalGap;
+  let left = targetRect.left - horizontalOffset;
+
+  if (top < editorScrollerRect.top) {
+    top += floatingElemRect.height + targetRect.height + verticalGap * 2;
+  }
+
+  if (left + floatingElemRect.width > editorScrollerRect.right) {
+    left = editorScrollerRect.right - floatingElemRect.width - horizontalOffset;
+  }
+
+  top -= anchorElementRect.top;
+  left -= anchorElementRect.left;
+
+  floatingElem.style.opacity = '1';
+  floatingElem.style.transform = `translate(${left}px, ${top}px)`;
+};
+
+interface FloatingLinkEditorProps {
+  editor: LexicalEditor;
+  isLink: boolean;
+  setIsLink: Dispatch<boolean>;
+  anchorElement: HTMLElement;
+}
+
+type LexicalSelection = RangeSelection | GridSelection | NodeSelection | null;
+
+const SUPPORTED_URL_PROTOCOLS = ['http:', 'https:'];
+
+const sanitizeUrl = (url: string) => {
+  try {
+    const parsedUrl = new URL(url);
+    if (!SUPPORTED_URL_PROTOCOLS.includes(parsedUrl.protocol)) {
+      return 'about:blank';
+    }
+  } catch {
+    return url;
+  }
+  return url;
+};
+
+// Source: https://stackoverflow.com/a/8234912/2013580
+const urlRegExp = new RegExp(
+  /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[\w]*))?)/,
+);
```

Comment:
Denne er hentet fra lexical. Burde vi lage en `validateUrl`-util og plassere den i `ndla/util`? Isåfall, hvordan skal vi håndtere whitelisting?